src/movies.js

```diff
@@ -1,28 +1,59 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
+
+const movies = require("./data");
+
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(mArray) {
+  const allDirectors = mArray.map(function (direct) {
+    return direct.director;
+  });
+  return allDirectors;
+}
+getAllDirectors(movies);
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(arrMov) {
+  let numbOfMov = arrMov.filter(function (movie) {
+    if (movie.director === 'Steven Spielberg' && movie.genre.includes('Drama'))
+      return movie;
+
+  });
+  return (numbOfMov.length);
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage() {
+
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore() { }
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear() {}
+function orderByYear(movieYear) {
+  const result = movieYear.sort((firstScore, secondScore) => {
+    if (firstScore.year > secondScore.year) {
+      return -1;
+    } else {
+      return 1;
```

Comment:
All logic is correct, you just mixed up the 1 with -1. 

function compare(a, b) {
  if (a is more than b) {
    return 1;
  }
  if (a is less than b) {
    return -1;
  }
}

Reply:
Thank you for your feedback:) it is really important for me.I will do it:)